packages/server/lib/browsers/bidi_automation.ts

```diff
@@ -0,0 +1,277 @@
+import debugModule from 'debug'
+import type { Automation } from '../automation'
+import type { BrowserPreRequest, BrowserResponseReceived, ResourceType } from '@packages/proxy'
+import type { Client as WebDriverClient } from 'webdriver'
+import type {
+  NetworkBeforeRequestSentParameters,
+  NetworkResponseStartedParameters,
+  NetworkResponseCompletedParameters,
+  NetworkFetchErrorParameters,
+  BrowsingContextInfo,
+} from 'webdriver/build/bidi/localTypes'
+const debug = debugModule('cypress:server:browsers:bidi_automation')
+const debugVerbose = debugModule('cypress-verbose:server:browsers:bidi_automation')
+
+// NOTE: these types will eventually be generated automatically via the 'webdriver' package
+// Taken from https://fetch.spec.whatwg.org/#request-initiator-type
+type RequestInitiatorType = 'audio' | 'beacon' | 'body' | 'css' | 'early-hints' | 'embed' | 'fetch' | 'font' | 'frame' | 'iframe' | 'image' | 'img' | 'input' | 'link' | 'object' | 'ping' | 'script' | 'track' | 'video' | 'xmlhttprequest' | 'other' | null
+// Taken from https://fetch.spec.whatwg.org/#concept-request-destination
+type RequestDestination = 'audio' | 'audioworklet' | 'document' | 'embed' | 'font' | 'frame' | 'iframe' | 'image' | 'json' | 'manifest' | 'object' | 'paintworklet' | 'report' | 'script' | 'serviceworker' | 'sharedworker' | 'style' | 'track' | 'video' | 'webidentity' | 'worker' | 'xslt' | ''
+
+export type NetworkBeforeRequestSentParametersModified = NetworkBeforeRequestSentParameters & {
+  request: {
+    destination: RequestDestination
+    initiatorType: RequestInitiatorType
+  }
+}
+
+// maps the network initiator to a ResourceType (which is initially based on CDP).
+// This provides us with consistency of types in our request/response middleware, which is important for cy.intercept().
+const normalizeResourceType = (type: RequestInitiatorType): ResourceType => {
```

Comment:
I can't remember, did we talk about whether or not this would be considered a breaking change? Like if someone is intercepting a certain resource type but the types are different now so the intercept breaks.

Reply:
We did briefly which was one of the rationales around deprecating the field. The idea that we are trying our best to map these 1:1 and the types be the same shouldn't make it a breaking change, but it's difficult to predict the behavior for each type. That being said I don't think its a breaking change but there could be bugs that we will need to fix. I feel confident on fetch and xhr types, but the fonts might be a little more nuanced which is in use as seen [here](https://github.com/cypress-io/cypress/issues/30447#issuecomment-2599865978). Worse case, if a bug is present with the mapping, the user can opt out with the `FORCE_FIREFOX_CDP` until we fix it